src/app/components/core/control-dispatcher/list-control-dispatcher/list-control-dispatcher.component.ts

```diff
@@ -34,6 +39,69 @@ export class ListControlDispatcherComponent implements OnInit {
       .subscribe(viewport => {
         this.viewPort = viewport;
       });
+    this.getInstalledServicesList();
+  }
+
+  public getInstalledServicesList() {
+    /** request start */
+    this.ngProgress.start();
+    this.servicesApiService.getInstalledServices().
+      then((data: any) => {
+        /** request done */
+        this.ngProgress.done();
+        // Check if dispatcher service available in installed services list
+        this.dispatcherServiceInstalled = data.services.includes('dispatcher');
+        if (this.dispatcherServiceInstalled) {
+          this.checkServiceStatus();
+        }
+      })
+      .catch(error => {
+        /** request done */
+        this.ngProgress.done();
+        console.log('service down ', error);
+      });
+  }
+
+  public checkServiceStatus() {
+    /** request start */
+    this.ngProgress.start();
+    this.servicesApiService.getAllServices()
+      .subscribe((res: any) => {
+        /** request done */
+        this.ngProgress.done();
+        this.dispatcherService = null; // reset service
+        // Check if dispatcher service is added
+        this.dispatcherService = res.services.find((svc: any) => svc.type === 'Dispatcher');
+        if (this.dispatcherService) {
+          this.getSchedules();
```

Comment:
Get schedules first.

You don't need to check service then; if enabled it can be seen after hovering on  icon for services status.